### DISCUSSION TITLE
How to connect Wakatime to starship over fish-shell

### DISCUSSION THREAD

**Discussion by weeebdev** [starship/starship#1410] [General]
By default fish-shell uses `~/.config/fish/functions/fish_prompt.fish` (with omf). But as far as I understand when I install starship it stops using fish_prompt.fish, since starship configs use only _.toml_, not _.fish_ where do I have to put [this](https://raw.githubusercontent.com/Cyber-Duck/fish-wakatime/master/fish-wakatime.sh) in order to use Wakatime in fish?

**Comment by matchai** [comment]
> **⚠️ Warning**
> This is likely to be fine for most updates to Starship, but can eventually be broken by rewrites of the initialization functionality in Starship.


For the time being, there's no way to add a custom script to the `fish_prompt` function generated by Starship, however, it should be possible to extend it.

1. Remove the Starship initialization line from `~/.config/fish/config.fish`.
1. Create a new `fish_prompt.fish` file with the one generated by the Starship init script:
    ```fish
    starship init fish --print-full-init > ~/.config/fish/functions/fish_prompt.fish
    ```
1.  Modify the newly created `~/.config/fish/functions/fish_prompt.fish` file to include the contents of the wakatime script.

**Reply by weeebdev** [reply]
> > **⚠️ Warning**
> > This is likely to be fine for most updates to Starship, but can eventually be broken by rewrites of the initialization functionality in Starship.
> 
> For the time being, there's no way to add a custom script to the `fish_prompt` function generated by Starship, however, it should be possible to extend it.
> 
> 1. Remove the Starship initialization line from `~/.config/fish/config.fish`.
> 2. Create a new `fish_prompt.fish` file with the one generated by the Starship init script:
>    ```fish
>    starship init fish --print-full-init > ~/.config/fish/functions/fish_prompt.fish
>    ```
> 3. Modify the newly created `~/.config/fish/functions/fish_prompt.fish` file to include the contents of the wakatime script.

It seems to me that it doesn't work anymore from v0.45.0 or higher

**Reply by weeebdev** [reply]
Sorry. Now it works
